Lib/test/test_subprocess.py

```diff
@@ -1705,16 +1705,15 @@ def test_start_new_session(self):
         # still indicates that it was called.
         try:
             output = subprocess.check_output(
-                    [sys.executable, "-c",
-                     "import os; print(os.getpgid(os.getpid()))"],
+                    [sys.executable, "-c", "import os; print(os.getsid(0))"],
                     start_new_session=True)
         except OSError as e:
             if e.errno != errno.EPERM:
                 raise
         else:
-            parent_pgid = os.getpgid(os.getpid())
```

Comment:
i'm wondering if there is a reason this code was calling os.getpgid() rather than os.getsid() but I honestly do not remember.  We added this test 9 years ago with the _posixsubprocess.c internals rewrite, regardless of that calling setsid().  My PGID vs SID knowledge is vague and needs refreshing, they're probably the same in this situation?

Reply:
I cannot tell you why you wrote this code :-)